Create class components to render the main user interface

luis-pomare opened the pull request:
# In this milestone, we have:
* Removed all boilerplate files and default styles created by the create-react-app toolchain
* Created the calculator component that renders the main user interface
* Added the layout to the calculator interface
* Added the style to match the given design
* Ran the eslint and stylelint to ensure the usage of good practice in code
* We have used only class-based components

luis-pomare committed:
Add calculator skelleton

BenMukebo commented on src/App.js:
```diff
@@ -1,29 +1,22 @@
-import logo from './logo.svg';
+import React from 'react';
 import './App.css';
+// eslint-disable-next-line
```
- [x] it is always a good practice to make sure you fix all linter errors instead of disabling them, A linter is great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. 💯 😉 

luis-pomare replied:
![image](https://user-images.githubusercontent.com/80157011/176185449-c5a9f449-e816-4d86-8954-65987427033d.png)

As you can see when I use the linters without those comments I have errors because for react is preferred to utilize function components instead of class components.
Since it is a core requirement for this project to use class components I have no other option than keep those disabling comments.

BenMukebo commented on src/App.js:
```diff
@@ -1,29 +1,22 @@
-      </header>
-    </div>
-  );
+// eslint-disable-next-line
```
- [x] it is always a good practice to make sure you fix all linter errors instead of disabling them, A linter is great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. 💯 😉 

luis-pomare replied:
![image](https://user-images.githubusercontent.com/80157011/176185449-c5a9f449-e816-4d86-8954-65987427033d.png)

As you can see when I use the linters without those comments I have errors because for react is preferred to utilize function components instead of class components.
Since it is a core requirement for this project to use class components I have no other option than keep those disabling comments.

BenMukebo commented on src/App.js:
```diff
@@ -1,29 +1,22 @@
-  );
+// eslint-disable-next-line
+class App extends React.Component {
+  // eslint-disable-next-line
```
- [x] it is always a good practice to make sure you fix all linter errors instead of disabling them, A linter is great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. 💯 😉 

luis-pomare replied:
![image](https://user-images.githubusercontent.com/80157011/176185449-c5a9f449-e816-4d86-8954-65987427033d.png)

As you can see when I use the linters without those comments I have errors because for react is preferred to utilize function components instead of class components.
Since it is a core requirement for this project to use class components I have no other option than keep those disabling comments.

BenMukebo commented on src/components/calculator.js:
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+// eslint-disable-next-line
+export default class Calculator extends React.Component {
+  // eslint-disable-next-line
```
- [x] it is always a good practice to make sure you fix all linter errors instead of disabling them, A linter is great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. 💯 😉 

luis-pomare replied:
![image](https://user-images.githubusercontent.com/80157011/176185449-c5a9f449-e816-4d86-8954-65987427033d.png)

As you can see when I use the linters without those comments I have errors because for react is preferred to utilize function components instead of class components.
Since it is a core requirement for this project to use class components I have no other option than keep those disabling comments.

BenMukebo reviewed (changes requested):
### Hi @luis-pomare,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- Gitflow was correctly used.  ✔️   
- The Calculator component  match the following given design ✔️ 
- Professional README ✔️ 
- All the boilerplate that got created by the CRA have gotten deleted. ✔️ 
- You've used class based component  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [x] Please, Write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. you can check the  [Professional repo - basic rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] Kindly, get used to committing your code, A Git Commit Message helps identify all the changes in the documentation. When code is committed often, GIT can actually be more useful and can help you know what changes you have made for each step towards the complete feature. I would commit often and early, especially if you're on a team. Lack of commitment can often cause problems in fixing errors and putting more work into fixing them.



Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

luis-pomare commented:
@BenMukebo Thank you very much for your help.
I have updated the title of the pull request and summarized the changes included in this milestone as has been requested in the first code review.
I have one doubt about the comments that disable the linter checks on the class components. I added those comments because one full-time code reviewer said to me that this is the only way of not having errors when using class components. I tried to run the linters without that comments and they mark the usage of react class components as an error, for that reason I scheduled one meeting with a full-time code reviewer and that was the solution that he gave me because it is a core requirement for this project to use class components.

luis-pomare committed:
Fix disabled linters

luis-pomare committed:
Add disabling comments again to avoid undesired linter errors

EricMbouwe reviewed (approved):
Hi @luis-pomare, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
